File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.java
import java.util.*;
class Solution {
    public String[] solution(String[] records) {

        HashMap<String,String> map = new HashMap<>();
        
        // map에 id와 nickname 저장
        for(String record : records){
            StringTokenizer st = new StringTokenizer(record);
            String command = st.nextToken();
            String id = st.nextToken();

            if(command.equals("Enter") || command.equals("Change")){
                String nickname = st.nextToken();
                map.put(id, nickname);
            }

        }
        
        // 채팅방 출력하기
        List<String> result = new ArrayList<>();
        for(String record : records){
            StringTokenizer st = new StringTokenizer(record);
            String command = st.nextToken();
            String id = st.nextToken();
            String nickname = map.get(id);

            if(command.equals("Enter")){
                result.add(nickname + "님이 들어왔습니다.");
            }else if(command.equals("Leave")){
                result.add(nickname + "님이 나갔습니다.");
            }
        }
    
        // list -> arr
        return result.toArray(String[]::new);
    }
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    
    nick_dict = {}
    
    for rec in record:
        t , *remains = rec.split()
        if t == "Enter" or t=="Change":
            u_id,nick = remains
            nick_dict[u_id] = nick
    result = []
    for rec in record:
        t,*remains = rec.split()
        
        if t=="Enter":
            k = remains[0]
            result.append(f"{nick_dict[k]}님이 들어왔습니다.")
        elif t =="Leave":
            k = remains[0]
            result.append(f"{nick_dict[k]}님이 나갔습니다.")
    
    return result